### DISCUSSION TITLE
How to delay snapshot creation until the network is connected (WiFi)?

### DISCUSSION THREAD

**Discussion by ryzen3100** [garethgeorge/backrest#602] [Q&A] [answered]
Is it possible to delay taking the snapshot until the network is active, or maybe wait a few seconds before starting the snapshot?
The problem is that when I wake up my computer from sleep, it sometimes takes 4 to 8 seconds to reconnect to WiFi, and Backrest gives an error because it can’t reach the repository.

**Comment by jwyse** [comment, marked as answer]
There's a Command Hook Example in the docs: [Add a hook to check for internet connectivity before running a backup](https://garethgeorge.github.io/backrest/cookbooks/command-hook-examples#check-for-internet-connectivity)

Edit: Instead of using the `ON_ERROR_CANCEL` error behavior in the example, use one of the `ON_ERROR_RETRY_` behaviors.  I haven't tested this but it seems pretty straightforward.

**Reply by ryzen3100** [reply]
Thank you it worked!
This is the snippet I am using in Windows.
```
function Test-InternetConnectivity {
  param()

  $pingResult = Test-Connection -ComputerName google.com -Count 1 -Quiet

  if ($pingResult) {
    Write-Output "Internet connection is up"
    exit 0
  } else {
    Write-Output "Internet connection is down"
    exit 1
  }
}

# Call the function
Test-InternetConnectivity
```
--- edit ---
Updated the script, moved the main part to a function otherwise it wouldn't work under Win11.
